src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/service/DataCleanerTest.java

```diff
@@ -26,67 +25,75 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
 
 import org.junit.Before;
-import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.ExpectedException;
 import org.junit.runner.RunWith;
-import org.mockito.Matchers;
+import org.mockito.ArgumentMatcher;
 import org.mockito.Mock;
+import org.mockito.Mockito;
 import org.mockito.runners.MockitoJUnitRunner;
 
 import com.google.api.gax.paging.Page;
-import com.google.cloud.RetryOption;
-import com.google.cloud.bigquery.BigQueryError;
-import com.google.cloud.bigquery.Job;
-import com.google.cloud.bigquery.JobStatus;
 import com.google.cloud.bigquery.Table;
 import com.google.cloud.bigquery.TableId;
 import com.google.cloud.storage.Blob;
 import com.google.cloud.storage.BlobId;
 import com.google.cloud.storage.Bucket;
-import com.google.cloud.storage.BucketInfo;
 import com.google.cloud.storage.Storage;
 import com.google.cloud.storage.StorageException;
 import com.google.common.collect.ImmutableList;
 
 import com.hotels.bdp.circustrain.api.CircusTrainException;
+import com.hotels.bdp.circustrain.bigquery.extraction.container.ExtractionContainer;
+import com.hotels.bdp.circustrain.bigquery.extraction.container.ExtractionUri;
+import com.hotels.bdp.circustrain.bigquery.extraction.service.DataCleaner;
 
 @RunWith(MockitoJUnitRunner.class)
-public class BigQueryDataExtractionServiceTest {
-
-  public @Rule ExpectedException expectedException = ExpectedException.none();
+public class DataCleanerTest {
 
+  private DataCleaner cleaner;
   private @Mock Storage storage;
   private @Mock Table table;
-  private @Mock Job job;
-  private @Mock JobStatus jobStatus;
-
-  private BigQueryDataExtractionService service;
+  private @Mock Future future;
+  private @Mock ExecutorService executorService;
 
   @Before
-  public void init() {
-    service = new BigQueryDataExtractionService(storage);
+  public void init() throws ExecutionException, InterruptedException {
+    initExecutor();
+    cleaner = new DataCleaner(storage);
   }
 
-  @Test
-  public void extract() throws InterruptedException {
-    BigQueryExtractionData data = new BigQueryExtractionData();
-    TableId tableId = TableId.of("dataset", "table");
-    when(table.extract(anyString(), anyString())).thenReturn(job);
-    when(table.getTableId()).thenReturn(tableId);
-    when(job.waitFor(Matchers.<RetryOption> anyVararg())).thenReturn(job);
-    when(job.getStatus()).thenReturn(jobStatus);
-    when(jobStatus.getError()).thenReturn(null);
-    service.extract(table, data);
-    verify(storage).create(any(BucketInfo.class));
-    verify(table).extract(eq("csv"), eq(data.getDataUri()));
+  private void initExecutor() {
+    Future future = mock(Future.class);
```

Comment:
You already have a mocked Future from line 62 so you should be able to get rid of this one.